src/v1/sources/shopify/webhookTransformations/serverSideUtlis.js

```diff
@@ -54,8 +58,73 @@ const getAnonymousIdFromAttributes = (event) => {
   return rudderAnonymousIdObject ? rudderAnonymousIdObject.value : null;
 };
 
+/**
+ * Returns the cart_token from the event message
+ * @param {Object} event
+ * @returns {String} cart_token
+ */
+const getCartToken = (event) => event?.cart_token || null;
+
+/**
+ * Handles the anonymousId assignment for the message, based on the event attributes and redis data
+ * @param {Object} message
+ * @param {Object} event
+ */
+const handleAnonymousId = async (message, event) => {
+  const anonymousId = getAnonymousIdFromAttributes(event);
+  if (isDefinedAndNotNull(anonymousId)) {
+    message.anonymousId = anonymousId;
+  } else {
+    // if anonymousId is not present in note_attributes or note_attributes is not present, query redis for anonymousId
+    const cartToken = getCartToken(event);
+    if (cartToken) {
+      const redisData = await RedisDB.getVal(`pixel:${cartToken}`);
+      if (redisData?.anonymousId) {
+        message.anonymousId = redisData.anonymousId;
+      }
+    }
+  }
+};
+
+/**
+  Handles userId, email and contextual properties enrichment for the message payload
+ * @param {Object} message
+ * @param {Object} event
+ * @param {String} shopifyTopic
+*/
+const handleCommonProperties = (message, event, shopifyTopic) => {
+  if (message.userId) {
+    message.userId = String(message.userId);
+  }
+  if (!get(message, 'traits.email')) {
+    const email = extractEmailFromPayload(event);
+    if (email) {
+      message.setProperty('traits.email', email);
+    }
+  }
+  message.setProperty(`integrations.${INTEGERATION}`, true);
+  message.setProperty('context.library', {
+    eventOrigin: 'server',
+    name: 'RudderStack Shopify Cloud',
+    version: '2.0.0',
+  });
+  message.setProperty('context.topic', shopifyTopic);
+  // attaching cart, checkout and order tokens in context object
+  message.setProperty(`context.cart_token`, event.cart_token);
+  message.setProperty(`context.checkout_token`, event.checkout_token);
+  // raw shopify payload passed inside context object under shopifyDetails
+  message.setProperty('context.shopifyDetails', event);
+  if (shopifyTopic === 'orders_updated') {
+    message.setProperty(`context.order_token`, event.token);
+  }
```

Comment:
why is this logic specifically being added?

Reply:
it was a request from CS long back I think, in the v0 shopify. they wanted these tokens within context

Reply:
but why only for `orders_updated`?

Reply:
Is token not always order_token across events ? If yes, if clause can be removed - whichever events has it, those gets added